### PR TITLE
Add media queries and mobile viewport

### PR DESCRIPTION
**Pull request by daleharvey**
As discussed in https://github.com/vector-im/riot-web/issues/9814

I made an additional PR to matrix-react-sdk @ https://github.com/matrix-org/matrix-react-sdk/pull/3991

### CODE REVIEW

**Comment by daleharvey** [comment]
Copying the comment over since guessing we want to keep most discussion here?

So there is a lot of work to do here, I wanted to put up the minimal patch that gets the UI usable on mobile and touches the least amount of things as possible so we can discuss the overall approach without too big a patch.

It looks like:

https://i.imgur.com/Z8TU0VE.png
https://i.imgur.com/gvVVlJu.png
https://i.imgur.com/jSMuqAN.png

I feel like we probably dont want to be repeating this media query so often so not sure if we want to put this all in a dedicated mobile-overrides.scss ?

**Comment by daleharvey** [comment]
Signed-off-by: Dale Harvey <dale@arandomurl.com>

**Comment by jryans** [comment]
Thanks for the contribution! 😄 As mentioned in few places, we're currently in crunch mode for FOSDEM prep, but we'll take a look at this next week.

**Comment by jryans** [comment]
Before moving forward here, I'd like discuss with the Riot core team on our support level for mobile web. I'll keep you updated, and sorry for the delays.

**Comment by turt2live** [comment]
(removing this from the github review queue while that happens)

**Comment by jryans** [comment]
Some progress to report here! https://github.com/vector-im/riot-web/pull/12377 intends to update our supported environments for Riot. Assuming that moves forward and merges, we'll be ready to pick this up for review.

**Comment by jryans** [comment]
The supported environments set has been updated, and so this can now proceed to review.

**Comment by jryans** [comment]
To clarify for reviewers, these are ready for anyone to review. https://github.com/vector-im/riot-web/issues/9814 is not on the workflow board itself since this set is only an initial step.

File: res/welcome.html [hunk 1]
```diff
@@ -158,6 +158,16 @@
     display: none;
 }
 
+@media only screen and (max-width : 480px) {
+  .mx_ButtonRow {
```
Comment (jryans) [inline res/welcome.html]: Please use 4 space indent.

File: res/welcome.html [hunk 2]
```diff
@@ -158,6 +158,16 @@
     display: none;
 }
 
+@media only screen and (max-width : 480px) {
```
Comment (jryans) [inline res/welcome.html]: Please remove the space before the colon.

**Comment by t3chguy** [comment]
Hello @daleharvey, thank you for your contribution a while back, Any chance you could just do the final small tweaks to get it over the line? If not then one of the core team may do so as this would be a nice thing to land.

Thanks

File: res/welcome.html (hunk 2 above)
Comment (t3chguy) [inline res/welcome.html]:
```suggestion
@media only screen and (max-width: 480px) {
```

File: res/welcome.html [hunk 3]
```diff
@@ -158,6 +158,16 @@
...
+  .mx_ButtonRow {
+    flex-direction: column;
+  }
+
+  .mx_ButtonRow > * {
+      margin: 0 0 10px 0;
+  }
+}
```
Comment (t3chguy) [inline res/welcome.html]:
```suggestion
    .mx_ButtonRow {
        flex-direction: column;
    }

    .mx_ButtonRow > * {
        margin: 0 0 10px 0;
    }
}
```

**Comment by TitanNano** [comment]
@jryans @t3chguy I made the requested changes. https://github.com/vector-im/riot-web/pull/13818

**Pull request closed without merging**